File: system-test/java/src/main/java/com/mycompany/myshop/testkit/driver/adapter/ui/client/pages/HomePage.java
package com.mycompany.myshop.testkit.driver.adapter.ui.client.pages;

import com.mycompany.myshop.testkit.driver.adapter.shared.client.playwright.PageClient;

public class HomePage extends BasePage {
    private static final String MY_SHOP_BUTTON_SELECTOR = "a[href='/new-order']";
    private static final String ORDER_HISTORY_BUTTON_SELECTOR = "a[href='/order-history']";
    private static final String COUPON_MANAGEMENT_BUTTON_SELECTOR = "a[href='/admin-coupons']";

    public HomePage(PageClient pageClient) {
        super(pageClient);
    }

    public NewOrderPage clickNewOrder() {
        pageClient.click(MY_SHOP_BUTTON_SELECTOR);
        return new NewOrderPage(pageClient);
    }

    public OrderHistoryPage clickOrderHistory() {
        pageClient.click(ORDER_HISTORY_BUTTON_SELECTOR);
        return new OrderHistoryPage(pageClient);
    }

    public CouponManagementPage clickCouponManagement() {
        pageClient.click(COUPON_MANAGEMENT_BUTTON_SELECTOR);
        return new CouponManagementPage(pageClient);
    }
}

File: system-test/dotnet/Driver.Adapter/Ui/Client/Pages/HomePage.cs
using Driver.Adapter.Shared.Client.Http;

using Driver.Adapter.Shared.Client.Playwright;



namespace Driver.Adapter.Ui.Client.Pages;



public class HomePage : BasePage

{

    private const string MyShopButtonSelector = "a[href='/new-order']";

    private const string OrderHistoryButtonSelector = "a[href='/order-history']";

    private const string CouponManagementButtonSelector = "a[href='/admin-coupons']";



    public HomePage(PageClient pageClient) : base(pageClient)

    {

    }



    public async Task<NewOrderPage> ClickNewOrderAsync()

    {

        await PageClient.ClickAsync(MyShopButtonSelector);

        return new NewOrderPage(PageClient);

    }



    public async Task<OrderHistoryPage> ClickOrderHistoryAsync()

    {

        await PageClient.ClickAsync(OrderHistoryButtonSelector);

        return new OrderHistoryPage(PageClient);

    }



    public async Task<CouponManagementPage> ClickCouponManagementAsync()

    {

        await PageClient.ClickAsync(CouponManagementButtonSelector);

        return new CouponManagementPage(PageClient);

    }

}





